aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp

```
@@ -204,5 +207,36 @@ Tensor NestedTensor_to_padded_tensor_cuda(
   }
   return NestedTensor_to_padded_tensor_generic(t, padding, output_size);
 }
+
+Tensor flash_scaled_dot_product_self_attention(
+    const Tensor& qkv,
+    const Tensor& cumulative_sequence_length,
+    const int64_t max_seqlen_batch,
+    double dropout_p,
+    bool causal) {
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11060 && !defined(_MSC_VER) && \
+    !(__APPLE__ || __MACH__)
+  auto q = qkv.index({at::indexing::Slice(), 0});
+  auto k = qkv.index({at::indexing::Slice(), 1});
+  auto v = qkv.index({at::indexing::Slice(), 2});
+  auto softmax_scale = std::pow(qkv.size(-1), -0.5);
+  std::vector<Tensor> output = fmha::mha_fwd(
+      q,
+      k,
+      v,
+      cumulative_sequence_length,
+      cumulative_sequence_length,
+      max_seqlen_batch,
+      max_seqlen_batch,
+      softmax_scale,
+      false,
+      causal,
+      false,
+      c10::nullopt);
+  return output[0];
+#endif
+  return Tensor{};
```

I'd throw an exception here in case it gets called by accident.